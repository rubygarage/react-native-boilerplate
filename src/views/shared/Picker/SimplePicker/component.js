import React from 'react';
import { Picker } from '@react-native-community/picker';
import PropTypes from 'prop-types';

import * as S from '../styled';

const SimplePicker = ({ onValueChange, selectedValue, options }) => (
  <S.PickerView
    onValueChange={onValueChange}
    selectedValue={selectedValue}
  >
    {options.map((item) => <Picker.Item {...item} />)}
  </S.PickerView>
);

SimplePicker.propTypes = {
  onValueChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
  selectedValue: PropTypes.string,
};

SimplePicker.defaultProps = {
  selectedValue: undefined,
};

export default SimplePicker;
