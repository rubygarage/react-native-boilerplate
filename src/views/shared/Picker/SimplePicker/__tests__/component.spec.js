import React from 'react';
import { shallow } from 'enzyme';

import SimplePicker from '../component';

describe('SimplePicker component', () => {
  const defaultProps = {
    onValueChange: jest.fn(),
    selectedValue: 'value1',
    options: [
      { key: 'key0', label: 'label0', value: 'value0' },
      { key: 'key1', label: 'label1', value: 'value1' }],
  };
  const component = shallow(<SimplePicker {...defaultProps} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
