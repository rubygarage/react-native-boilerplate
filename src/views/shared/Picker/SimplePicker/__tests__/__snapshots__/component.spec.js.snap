// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePicker component renders correctly 1`] = `
<Styled(Picker)
  onValueChange={[MockFunction]}
  selectedValue="value1"
>
  <PickerItem
    key="key0"
    label="label0"
    value="value0"
  />
  <PickerItem
    key="key1"
    label="label1"
    value="value1"
  />
</Styled(Picker)>
`;
